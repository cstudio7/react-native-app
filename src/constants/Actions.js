export default {
  favoriteName: 'FAVORITE_NAME',
  unfavoriteName: 'UNFAVORITE_NAME',
  saveNames: 'SAVE_NAMES'
};