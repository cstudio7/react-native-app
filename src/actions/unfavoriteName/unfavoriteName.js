import Actions from '../../constants/Actions';

export default (gender, name) => ({
  type: Actions.unfavoriteName,
  gender,
  name
});
