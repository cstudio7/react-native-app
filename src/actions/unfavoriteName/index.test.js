import unfavoriteName from './';
import Actions from '../../constants/Actions';

const mockedName = Object.freeze({ name: 'artur', isFavoite: true });
const mockedNames = Object.freeze(new Array(mockedName));

it('creates a UNFAVORITE_NAME action', () => {
  expect(unfavoriteName('female', mockedName)).toEqual({
    type: Actions.unfavoriteName,
    gender: 'female',
    name: mockedName
  });
  expect(unfavoriteName('female', mockedName)).toMatchSnapshot();
});
