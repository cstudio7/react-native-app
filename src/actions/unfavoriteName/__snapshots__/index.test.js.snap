// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creates a UNFAVORITE_NAME action 1`] = `
Object {
  "gender": "female",
  "name": Object {
    "isFavoite": true,
    "name": "artur",
  },
  "type": "UNFAVORITE_NAME",
}
`;
